R/utils.R

```diff
@@ -589,6 +589,7 @@ make_matrix_labels <- function(mat,lab,diag=TRUE) {
 
 # nocov start
 # TODO: give up on this
+#' @export
```

Comment:
I'm worried about exposing an `is.numeric` method for a non-mrgsolve object, especially something as core as `data.frame`. This could lead to surprising breakage in scripts that have loaded mrgsolve.

```R
d <- data.frame(
  foo = 1:3,
  bar = 3:5
)

sloop::s3_dispatch(is.numeric(d))
#>    is.numeric.data.frame
#>    is.numeric.default
#> => is.numeric (internal)
is.numeric(d)
#> [1] FALSE

loadNamespace("mrgsolve")
#> <environment: namespace:mrgsolve>

sloop::s3_dispatch(is.numeric(d))
#> => is.numeric.data.frame
#>    is.numeric.default
#>  * is.numeric (internal)
is.numeric(d)
#>  foo  bar
#> TRUE TRUE
```

In the long run, I think it'd be better to rename (or delete) `is.numeric.data.frame` and audit `is.numeric` call sites in mrgsolve for those that need to be adjusted.

<details>

<summary>call sites</summary>

**is.numeric**

```
$ git grep -Fn is.numeric
R/class_matlist.R:26:  x2 <- all(vapply(object@data, is.numeric, TRUE))
R/data_set.R:425:    nu <- sapply(idata, is.numeric)
R/env.R:32:  if(is.numeric(seed)) set.seed(seed)
R/events.R:127:  if(is.numeric(tinf) && length(tinf) > 0) l[["tinf"]] <- tinf
R/events.R:128:  if(is.numeric(until) && length(until) > 0) l[["until"]] <- until
R/events.R:159:    if(!is.numeric(ID)) {
R/events.R:164:      if(any(!is.numeric(data))) {
R/events.R:623:    # TODO: refactor once is.numeric is handled
R/events.R:624:    spacer <- is.atomic(evs[[i]]) && is.numeric(evs[[i]]) && length(evs[[i]])==1
R/events.R:694:  if(is.numeric(ID)) {
R/handle_spec_block.R:333:  if(is.numeric(number)) {
R/matlist.R:477:  if(!is.numeric(range)) {
R/modspec.R:78:    resolves_int <- is.numeric(etan) && all.equal(etan, round(etan))
R/mrgindata.R:279:  if(is.numeric(x)) {
R/mrgsolve.R:266:  if(is.numeric(nid) && !have_idata && !have_data) {
R/mrgsolve.R:584:      drop <- names(which(!is.numeric(join_data)))
R/mrgsolve.R:594:      drop <- names(which(!is.numeric(join_idata)))
R/mrgsolve.R:769:      if(!all(sapply(out[["data"]], is.numeric))) {
R/nmxml.R:133:  if(!is.numeric(index)) {
R/nmxml.R:402:  if(!is.numeric(index)) wstop("index did not resolve to a numeric value")
R/utils.R:434:single.number <- function(x) length(x)==1 & is.numeric(x)
R/utils.R:435:bare_numeric <- function(x) is.numeric(x) && !is.object(x)
R/utils.R:592:is.numeric.data.frame <- function(x) vapply(x,is.numeric,TRUE)
tests/testthat/test-data_set.R:138:  expect_true(all(mrgsolve:::is.numeric.data.frame(df)))
tests/testthat/test-data_set.R:142:  expect_true(all(mrgsolve:::is.numeric.data.frame(df)))
```

**single.number** (calls `is.numeric`)

```
$ git grep -Fn single.number
R/class_numericlist.R:22:  x1 <- all(sapply(object@data,single.number))
R/update.R:337:  non_nu <- !vapply(.y, FUN = single.number, TRUE)
R/utils.R:434:single.number <- function(x) length(x)==1 & is.numeric(x)
```

**bare\_numeric** (calls `is.numeric`)

```
$ git grep -Fn bare_numeric
R/mrgindata.R:61:  nu <- vapply(x, bare_numeric, TRUE)
R/utils.R:435:bare_numeric <- function(x) is.numeric(x) && !is.object(x)
```

</details>

Reply:
You're right. Thanks for pulling all of this out. 

I put a comment below to triage all of the call sites. This pattern is so man places, I decided to keep a function to do this (`numeric_columnsI()`) and subbed that in where ever I found the pattern (I believe).

Reply:
> I put a comment below to triage all of the call sites.

Thanks, that annotated list was very helpful.